clar2wasm/src/error_mapping.rs

```diff
@@ -0,0 +1,157 @@
+use clarity::vm::errors::{CheckErrors, Error, RuntimeErrorType, ShortReturnType, WasmError};
+use clarity::vm::types::ResponseData;
+use clarity::vm::Value;
+use wasmtime::{AsContextMut, Instance, Trap};
+
+const LOG2_ERROR_MESSAGE: &str = "log2 must be passed a positive integer";
+const SQRTI_ERROR_MESSAGE: &str = "sqrti must be passed a positive integer";
+const POW_ERROR_MESSAGE: &str = "Power argument to (pow ...) must be a u32 integer";
+
+pub enum ErrorMap {
+    ArithmeticOverflow = 0,
+    ArithmeticUnderflow = 1,
+    DivisionByZero = 2,
+    ArithmeticLog2Error = 3,
+    ArithmeticSqrtiError = 4,
+    UnwrapFailure = 5,
+    Panic = 6,
+    ShortReturnAssertionFailure = 7,
+    ArithmeticPowError = 8,
+    NotMapped = 99,
+}
+
+impl ErrorMap {
+    pub fn from(error_code: i32) -> ErrorMap {
+        match error_code {
+            0 => ErrorMap::ArithmeticOverflow,
+            1 => ErrorMap::ArithmeticUnderflow,
+            2 => ErrorMap::DivisionByZero,
+            3 => ErrorMap::ArithmeticLog2Error,
+            4 => ErrorMap::ArithmeticSqrtiError,
+            5 => ErrorMap::UnwrapFailure,
+            6 => ErrorMap::Panic,
+            7 => ErrorMap::ShortReturnAssertionFailure,
+            8 => ErrorMap::ArithmeticPowError,
+            _ => ErrorMap::NotMapped,
+        }
+    }
```

Comment:
No idea how we have missed that before, but shouldn't there be an instance of `From<i32>` instead of a method?

Also, I don't think `-1`, the default value for `error_code` should return `NotMapped`. If we have `-1`, it means that the runtime error is not a Clarity runtime error, but it should still be a defined error.

Reply:
> No idea how we have missed that before...

Yep, the implementations are pretty similar. Fixed.